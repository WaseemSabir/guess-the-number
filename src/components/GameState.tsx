import React from 'react';

interface GameStateProps {
    lower: string,
    upper: string,
    lastGuess: string,
    status: string | null
}

function GameState(props: GameStateProps) {
    const {lower, upper, lastGuess, status} = props

    return (
        <>
            <h3>Play!</h3>
            <div>Guess the number between {lower} and {upper}</div>
            <div>Last guess: {lastGuess}</div>
            {
                status && <div>{status}</div>
            }
        </>
    );
}

export default GameState;
