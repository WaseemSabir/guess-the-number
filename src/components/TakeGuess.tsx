import React, {useState} from 'react';

interface TakeGuessProps {
    onNewGuess: (new_guess: number) => void
}

function TakeGuess(props: TakeGuessProps) {
    const {onNewGuess} = props
    const [guess, setGuess] = useState('')

    function onMakeGuess() {
        let guess_num = Number(guess);
        onNewGuess(guess_num);
        setGuess('');
    }

    function onKeyDownInInput(event: React.KeyboardEvent) {
        if (event.key === 'Enter') {
            onMakeGuess()
        }
    }

    return (
        <>
            <label>Guess: </label>
            <input
                type='number'
                value={guess}
                onChange={e => setGuess(e.target.value)}
                onKeyDown={e => onKeyDownInInput(e)}
            />
            <button onClick={onMakeGuess}>Make guess</button>
        </>
    );
}

export default TakeGuess;
