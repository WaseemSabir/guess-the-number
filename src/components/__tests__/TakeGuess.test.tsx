import {cleanup, render} from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import GameState from "../GameState";
import TakeGuess from "../TakeGuess";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";


afterEach(cleanup)
Enzyme.configure({adapter: new Adapter()});

it('should render without crashes', () => {
    const {unmount} = render(getTakeGuess());
    unmount();
});

it('should match the last snapshot', () => {
    const component = renderer.create(
        getTakeGuess()
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("Should trigger newGuess on make guess button click", () => {
    const baseProps = {
        onNewGuess: jest.fn()
    }
    const wrapper = shallow(getTakeGuess(baseProps));
    wrapper.find("input").simulate('change', {target: {value: 100}});
    wrapper.find("button").simulate('click')
    expect(baseProps.onNewGuess).toHaveBeenCalledTimes(1);
    expect(baseProps.onNewGuess).toBeCalledWith(100);
});


function getTakeGuess(props: any = {}) {
    let {onNewGuess} = props

    onNewGuess = onNewGuess ? onNewGuess : (v: number) => {
    }

    return (
        <TakeGuess onNewGuess={onNewGuess}/>
    );
}
