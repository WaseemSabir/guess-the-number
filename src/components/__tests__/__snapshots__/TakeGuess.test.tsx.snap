// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the last snapshot 1`] = `
Array [
  <label>
    Guess: 
  </label>,
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    type="number"
    value=""
  />,
  <button
    onClick={[Function]}
  >
    Make guess
  </button>,
]
`;
